feat(sdk): change ex.table.get() method and create ex.table.tryGet() method

wzslr321 opened the pull request:
fixes #3879

- Change `ex.table.get()` method to throw an error instead of returning `{}` if no row with given key exists.
- Create `ex.table.tryGet()` method to return undefined when no row with the given key exists.

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [x] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

revitalbarletz commented on libs/wingsdk/src/shared-aws/table.inflight.ts:
```diff
@@ -67,7 +67,19 @@ export class TableClient implements ITableClient {
+  }
+
+  public async tryGet(key: string): Promise<Json | undefined> {
+    const command = new GetItemCommand({
```
No chance that an exception is thrown if key does not exist ? In other words, this.client.send will not throw an exception? I think that the answer is no based on the previous implementation of get method, but asking just in case.

tsuf239 commented on libs/wingsdk/src/shared-aws/table.inflight.ts:
```diff
@@ -67,7 +67,19 @@ export class TableClient implements ITableClient {
+    if (result.Item) {
+      return unmarshall(result.Item) as Json;
+    }
+    return undefined;
```
Can you please reuse this code in the get code?

tsuf239 commented on libs/wingsdk/test/shared-aws/table.inflight.test.ts:
```diff
@@ -96,24 +96,64 @@ describe("inflight table tests", () => {
+
+    // THEN
+    expect(response).toEqual(undefined);
+  });
```
Nice!

tsuf239 reviewed (commented):
Great job! Can you please add the tryGet method to the wing SDK spec tests (a collection of examples written in wing, we test both locally and on aws cloud) on `examples/tests/sdk_tests/table`- it can be a new file, or an addition to the current test files :)

wzslr321 commented:
Thank you for the review @tsuf239, because of your comment about SDK spec tests I've discovered a small missing piece of code regarding declaration of inflight methods! I feel much more confident about SDK parts of the codebase now :)

Changes since last review:
- Add tests inside `examples/tests/sdk_tests/table` | [here](https://github.com/winglang/wing/pull/4108/files#diff-6c79387d840d1dd5120e8a4888876dc799b92c216b2b550ae312d67ed856059aR1)
- Fill information about new `tryGet` inflight method | [here](https://github.com/winglang/wing/pull/4108/files#diff-389b05348f2430057d867fb12c22a67d7101595d95f3400e5e4a19b0ea82bc31R114)
- Reuse code for `get` and `tryGet` methods. | [here](https://github.com/winglang/wing/pull/4108/files#diff-160474b35b6800a633308bc92e56d9c8bca34bd0cd6ddbca660e2e83d5266baeR108)
- Revert incorrect changes pointed out in [this](https://github.com/winglang/wing/pull/4108#discussion_r1321175319) review comment

tsuf239 commented on libs/wingsdk/test/shared-aws/table.inflight.test.ts:
```diff
@@ -96,24 +96,64 @@ describe("inflight table tests", () => {
     };
+    dynamoMock
+      .on(GetItemCommand, expectedRequest)
+      .rejects(new Error("Row does not exist"));
```
I think there is only this left
